mesonbuild/scripts/__init__.py

```diff
@@ -12,10 +12,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from pathlib import Path
+
 def destdir_join(d1: str, d2: str) -> str:
     if not d1:
         return d2
+    d2_path = Path(d2)
+    d2_parts = d2_path.parts
     # c:\destdir + c:\prefix must produce c:\destdir\prefix
-    if len(d2) > 1 and d2[1] == ':':
-        return d1 + d2[2:]
-    return d1 + d2
+    if d2_path.drive or d2_path.root:
+        d2_parts = d2_parts[1:]
+    return str(Path(d1, *d2_parts))
```

Comment:
The API contract of destdir_join is that `d2` must always be an abspath, and if any installation path is not an abspath, it is prefixed with the core option "--prefix" (which itself is a fatal error if it isn't an abspath) before being passed into destdir_join.

So `d2_path.root` cannot ever be false; this check is not needed.

```suggestion
    d2_parts = Path(d2).parts[1:]
    return str(Path(d1, *d2_parts))
```